apps/judicial-system/web/src/routes/Court/Indictments/CourtRecord/CourtRecord.tsx

```diff
@@ -13,57 +17,63 @@ import {
   PageLayout,
   PageTitle,
   PdfButton,
+  UserContext,
 } from '@island.is/judicial-system-web/src/components'
-import { CourtSessionResponse } from '@island.is/judicial-system-web/src/graphql/schema'
 import { useCourtSessions } from '@island.is/judicial-system-web/src/utils/hooks'
 import { isIndictmentCourtRecordStepValid } from '@island.is/judicial-system-web/src/utils/validate'
 
 import CourtSessionAccordionItem from './CourtSessionAccordionItem'
 
 const CourtRecord: FC = () => {
-  const { workingCase, setWorkingCase, isLoadingWorkingCase, caseNotFound } =
+  const { user } = useContext(UserContext)
+  const { workingCase, isLoadingWorkingCase, caseNotFound, refreshCase } =
     useContext(FormContext)
-  const { createCourtSession, updateCourtSession } = useCourtSessions()
+  const { createCourtSession } = useCourtSessions()
   const [expandedIndex, setExpandedIndex] = useState<number>()
 
+  const hasGeneratedCourtRecord = hasGeneratedCourtRecordPdf(
+    workingCase.state,
+    workingCase.indictmentRulingDecision,
+    workingCase.withCourtSessions,
+    workingCase.courtSessions,
+    user,
+  )
+
   const handleNavigationTo = useCallback(
     (destination: string) => router.push(`${destination}/${workingCase.id}`),
     [workingCase.id],
   )
 
-  const stepIsValid = isIndictmentCourtRecordStepValid(
-    workingCase.courtSessions,
-  )
-
-  const handleConfirmClick = (
```

Comment:
Court session confirmation has been moved to the court session accordion item where all other court session updates are handled.